api/v1beta1/openstackprovisionserver.go

```diff
@@ -40,7 +40,7 @@ func AssignProvisionServerPort(
 	instance *OpenStackProvisionServer,
 	portStart int32,
 ) error {
-	if instance.Spec.Port != 0 {
```

Comment:
Not sure I understand. AssignProvisionServerPort() is called from baremetalset controller[1] and not from provisionserver controller. This is done when provisonserver is created/patched.

[1] https://github.com/openstack-k8s-operators/openstack-baremetal-operator/blob/master/controllers/openstackbaremetalset_controller.go#L474-L481

Reply:
I agree with @rabi here.  This `AssignProvisionServerPort` function is only used by the `OpenStackBarmetalSet` controller in the process of creating an `OpenStackProvisionServer` if one was not provided in the CR's `spec.provisionServerName` field.  The `OpenStackProvisionServer.spec.port` field needs to be provided and it will fail API validation if it is empty.